--- a/drivers/net/wireless/realtek/rtl8189fs/os_dep/linux/ioctl_cfg80211.c
+++ b/drivers/net/wireless/realtek/rtl8189fs/os_dep/linux/ioctl_cfg80211.c
@@ -17,6 +17,7 @@
 #include <drv_types.h>
 #include <rtw_wifi_regd.h>
 #include <wifi.h>
+#include <linux/version.h>

 ...

@@ -419,7 +420,11 @@ rtw_cfg80211_ch_switch_notify(_adapter *adapter, u8 ch, u8 bw, u8 offset, u8 ht)
         struct cfg80211_chan_def chdef;
         rtw_chdef_create(&chdef, wiphy, ch, bw, offset);

-        cfg80211_ch_switch_notify(adapter->pnetdev, &chdef, 0, 0);
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0)
+        cfg80211_ch_switch_notify(adapter->pnetdev, &chdef);
+#else
+        cfg80211_ch_switch_notify(adapter->pnetdev, &chdef, 0, 0);
+#endif

 };
